### PR TITLE
Added feature JSQVideoItem Thumbnail Property? #1220

### PR DESCRIPTION
**Pull request by aa-wong**
## Pull request checklist

- [ :heavy_check_mark: ] All tests pass. Demo project builds and runs.
- [ :heavy_check_mark: ] I have resolved any merge conflicts.
- [ :heavy_check_mark: ] I have squashed my commits into 1 commit.
- [ :heavy_check_mark: ] I have followed the [coding style](https://github.com/jessesquires/HowToContribute#style-guidelines), and reviewed the [contributing guidelines](https://github.com/jessesquires/JSQMessagesViewController/blob/develop/.github/CONTRIBUTING.md). Confirmation: :muscle::sunglasses::facepunch:

#### This fixes issue #1220.
Created a thumbnail feature on JSQVideoItem being able to encode/decode, init, and manual
## What's in this pull request?
thumbnail feature on JSQVideoItem
>close

### CODE REVIEW

**Comment by jessesquires** [comment]
Thanks @aa-wong ! :tada: 

Looks great at a glance! I'll have more feedback soon when I can take more time to review. :smile:

**Comment by benjaminhallock** [comment]
Lol we will never get this.

**Comment by benjaminhallock** [comment]
Also this version is poorly done. You removed the one line of code for legacy init, and you did not consider different resolutions when placing the play icon.

File: JSQMessagesViewController/Model/JSQVideoMediaItem.h [hunk 1]
```diff
@@ -32,8 +32,13 @@
 @property (nonatomic, strong) NSURL *fileURL;
 
 /**
+ *  The UIImage that creates video thumbnail.
```
Comment (jessesquires) [inline JSQMessagesViewController/Model/JSQVideoMediaItem.h]: should read:

> The thumbnail image for the video.

File: JSQMessagesViewController/Model/JSQVideoMediaItem.h [hunk 2]
```diff
@@ -51,6 +56,6 @@
  *  isReadyToPlay. Once the video has been saved to disk, or is ready to stream, you can
  *  set the fileURL property or isReadyToPlay property, respectively.
  */
-- (instancetype)initWithFileURL:(NSURL *)fileURL isReadyToPlay:(BOOL)isReadyToPlay;
+- (instancetype)initWithFileURL:(NSURL *)fileURL withThumbnail:(UIImage*)thumbnail isReadyToPlay:(BOOL)isReadyToPlay;
```
Comment (jessesquires) [inline JSQMessagesViewController/Model/JSQVideoMediaItem.h]: - we need to leave the original initializer.
- your new init should be:

```objective-c
- (instancetype)initWithFileURL:(NSURL *)fileURL isReadyToPlay:(BOOL)isReadyToPlay thumbnail:(UIImage *)thumbnail;
```

- then the original init should call this one with a `nil` thumbnail.
- need docs for the new initializer

File: JSQMessagesViewController/Model/JSQVideoMediaItem.m [hunk 3]
```diff
@@ -84,11 +91,26 @@ - (UIView *)mediaView
...
         UIImage *playIcon = [[UIImage jsq_defaultPlayImage] jsq_imageMaskedWithColor:[UIColor lightGrayColor]];
         
-        UIImageView *imageView = [[UIImageView alloc] initWithImage:playIcon];
+        UIImageView *imageView;
+        
+        if (self.thumbnail) {
+            UIImageView * iconView = [[UIImageView alloc] initWithImage:playIcon];
+            iconView.backgroundColor = [UIColor clearColor];
+            iconView.frame = CGRectMake(0.0f, 0.0f, size.width, size.height);
+            iconView.contentMode = UIViewContentModeCenter;
```
Comment (jessesquires) [inline JSQMessagesViewController/Model/JSQVideoMediaItem.m]: we should `.aspectFill` here

**Comment by jessesquires** [comment]
Thanks @aa-wong  !

We included your changes in #1823

**Pull request closed without merging**